x-pack/plugins/enterprise_search/public/applications/app_search/components/result/result.tsx

```diff
@@ -52,10 +53,13 @@ export const Result: React.FC<Props> = ({
     if (schemaForTypeHighlights) return schemaForTypeHighlights[fieldName];
   };
 
-  const documentLink = generatePath(ENGINE_DOCUMENT_DETAIL_PATH, {
-    engineName: resultMeta.engine,
-    documentId: resultMeta.id,
-  });
+  const documentLink = generatePath(
+    ENGINE_DOCUMENT_DETAIL_PATH,
+    encodePathParams({
+      engineName: resultMeta.engine,
+      documentId: resultMeta.id,
+    })
+  );
```

Comment:
Instead of using the default RR `generatePath`, I'm tempted to convert this to
```js
generateEnginePath(ENGINE_DOCUMENT_DETAIL_PATH, {
  engineName: resultMeta.engine,
  documentId: resultMeta.id,
})
```
Just so we get the encoding baked in without having to manually call it. I know Jason mentioned earlier that was maybe confusing though 🤷 What do you all think?

Reply:
I left another comment about a more generic `generateAppPath` that I think applies here.

Reply:
My objection still stands. It's like we're trying to force things through this state based helper that don't make sense to be part of a state based helper.

That being said, it's also inconsequential to me. It will be just fine if you choose to do this. I stated my objection, but either way will be just fine.

Reply:
Ya I think you're right about that, I shouldn't be conflating the two different helpers. I'm good with a generic `generateAppPath` (or maybe `generateEncodedPath` if it should be specific to encoding?), will add that helper to this  PR and have `generateEnginePath` call that helper.

Reply:
68a3057 & a72aef2